R/broom_wrapper.R

```diff
@@ -450,6 +454,95 @@ prediction <- function(df, data = "training", data_frame = NULL, conf_int = 0.95
   dplyr::group_by(ret, !!!rlang::syms(grouping_cols))
 }
 
+#' prediction wrapper for both training and test data
+#' @param df Data frame to predict. This should have model column.
+#' @export
+prediction_training_and_test <- function(df, prediction_type="default", threshold = 0.5, ...) {
+  test_index <- df %>% dplyr::filter(!is.null(.test_index)) %>% `[[`(1, ".test_index", 1)
+
+  grouped_cols <- colnames(df)[!colnames(df) %in% c("model", ".test_index", "source.data", ".model_metadata")]
+
+  tra_ret <- switch(prediction_type,
+                    default = prediction(df, ...),
+                    binary = prediction_binary(df, threshold = threshold, ...),
+                    conf_mat = prediction_binary(df, threshold = threshold, ...),
+                    coxph = prediction_coxph(df, ...))
+  tra_ret <- tra_ret %>% dplyr::mutate(is_test_data = FALSE) %>%
+                     dplyr::select(-is_test_data, everything(), is_test_data)
+  ret <- tra_ret
+
+  if (length(test_index) > 0) {
+    target_df <- if (length(grouped_cols) > 0) {
+      # like CAR RIER
+      # To avoid errors when passing CAR RIER-like columns to group_by_ in Standard Evaluation,
+      # enclose the column names in back quotes
+      df %>% dplyr::ungroup() %>% dplyr::group_by_(paste0('`', grouped_cols,'`'))
+    } else {
+      df
+    }
+
+    each_func <- function(df){
+      tryCatch({
+        if (!is.data.frame(df)) {
+          df <- tribble(~model,   ~.test_index,   ~source.data,
```

Comment:
How about adding tibble:: prefix just to be safe?
Also there are too many scripts between arguments here.